### ISSUE TITLE
Throw an exception instead of returning invalid values when determining platform and host triple

### ISSUE THREAD

**Issue by jdm** [servo/servo#16171]
https://github.com/servo/servo/issues/12356#issuecomment-289822444 is an example of a bootstrap operation failing because the msvc environment wasn't set up like mach expected. This caused us to try to fetch a nonexistent rustc build, and the resulting 404 error is not illuminating at all. We should be more aggressive about raising errors before we reach this point.

**Comment by nox** [comment]
cc @wafflespeanut

**Comment by jschwe** [comment]
I don't think this is an issue, anymore. fetching rust should completely be handled by rustup now.
